tools/validation-generator/src/main/java/io/spine/tools/validate/field/FieldValidatorFactories.java

```diff
@@ -0,0 +1,130 @@
+import static io.spine.tools.validate.field.FieldCardinality.SINGULAR;
+import static java.util.Optional.empty;
+
+/**
+ * A factory of {@link FieldValidatorFactory}s.
+ */
+public final class FieldValidatorFactories {
+
+    private final Expression<?> messageAccess;
+
+    /**
+     * Creates a new {@code FieldValidatorFactories}.
+     *
+     * @param messageAccess an expression which evaluates into the validated message value
+     */
+    public FieldValidatorFactories(Expression<?> messageAccess) {
+        this.messageAccess = checkNotNull(messageAccess);
+    }
+
+    /**
+     * Creates a new {@link FieldValidatorFactory} for the given field.
+     *
+     * @param field
+     *         the field to validate
+     * @return validator factory
+     */
+    public FieldValidatorFactory forField(FieldDeclaration field) {
+        return field.isCollection()
+               ? forCollections(field)
+               : forSingularField(field);
+    }
+
+    private FieldValidatorFactory forCollections(FieldDeclaration field) {
+        FieldValidatorFactory singularFactory =
+                forSingularField(field, REPEATED, CollectionFieldValidatorFactory.element);
+        String suffix = field.isMap() ? "Map().values()" : "List()";
+        Expression fieldAccess =
+                formatted("%s.get%s%s", messageAccess, field.name().toCamelCase(), suffix);
+        return new CollectionFieldValidatorFactory(field, fieldAccess, singularFactory);
+    }
+
+    private FieldValidatorFactory
+    forSingularField(FieldDeclaration field) {
+        Expression fieldAccess =
+                formatted("%s.get%s()", messageAccess, field.name().toCamelCase());
+        return forSingularField(field, SINGULAR, fieldAccess);
+    }
+
+    private static FieldValidatorFactory
+    forSingularField(FieldDeclaration field, FieldCardinality cardinality, Expression fieldAccess) {
+        JavaType type = field.isMap()
+                        ? field.valueDeclaration().javaType()
+                        : field.javaType();
+        switch (type) {
+            case STRING:
+                return new StringFieldValidatorFactory(field, fieldAccess, cardinality);
+            case INT:
+            case LONG:
+            case FLOAT:
+            case DOUBLE:
+                return new NumberFieldValidatorFactory(field, type, fieldAccess, cardinality);
+            case ENUM:
+            case MESSAGE:
+                return new MessageFieldValidatorFactory(field, fieldAccess, cardinality);
+            case BYTE_STRING:
+                return new ByteStringFieldValidatorFactory(field, fieldAccess, cardinality);
+            case BOOLEAN:
+            default:
+                return NoOpFactory.INSTANCE;
+        }
+    }
+
+    /**
+     * A {@link FieldValidatorFactory} which never produces any field validation.
+     */
+    private enum NoOpFactory implements FieldValidatorFactory {
```

Comment:
Please pull it to the package level. It would make the null object more visible for the people exploring the package.